### DISCUSSION TITLE
next/font/google does not create CSS Variables after refresh

### DISCUSSION THREAD

**Discussion by nareshbhatia** [vercel/next.js#59500] [Help] [answered]
### Summary

Very weird issue. I am using `next/font/google` to get fonts using CSS Variables. In dev mode, the first render after starting the server works perfectly - fonts are loaded and shown correctly. If I refresh the browser, fonts are no longer loaded because the CSS Variables are missing.

In production mode, fonts are loaded correctly even after refresh because CSS Variables exist even afte refresh.

Next.js version: 13.5.6

### First load
![image](https://github.com/vercel/next.js/assets/1220198/70d6739a-2d1f-41bf-8ad6-47dad4106732)

### After refresh
CSS Variables are lost

![image](https://github.com/vercel/next.js/assets/1220198/38d6d650-d9b1-4d01-afac-dabbcafd7449)

### Additional information

```js
import { Inter, Roboto_Mono as RobotoMono } from 'next/font/google';

const inter = Inter({
  subsets: ['latin'],
  variable: '--font-family-sans',
  display: 'swap',
});
const robotoMono = RobotoMono({
  subsets: ['latin'],
  variable: '--font-family-mono',
  display: 'swap',
});

interface RootLayoutProps {
  children: React.ReactNode;
}

export default function RootLayout({ children }: RootLayoutProps) {
  return (
    <html
      className={`${inter.variable} ${robotoMono.variable} theme-light`}
      lang="en"
    >
      <body className="bg-base text-primary antialiased">
        Hello World
      </body>
    </html>
  );
}
```

**Comment by nareshbhatia** [comment]
I found that CSS Variables are not lost, they are struck out, i.e. overridden by something else! However, I don't know who is overriding them. Any ideas?

![image](https://github.com/vercel/next.js/assets/1220198/a55401dd-722e-446b-818b-163cc3dd8528)

**Reply by ackvf** [reply]
So I actually have a problem where I don't have the variables made, but overlapping `font-family` definitions. What the heck?
<img width="294" height="320" alt="image" src="https://github.com/user-attachments/assets/f51b965c-539c-4d19-99d4-92da20166bcc" />

```
const inter = Inter({
  variable: "--font-inter",
  subsets: ["latin"],
})

const interTight = Inter_Tight({
  variable: "--font-inter-tight",
  subsets: ["latin"],
})

const robotoMono = Roboto_Mono({
  variable: "--font-roboto-mono",
  subsets: ["latin"],
  weight: ["400", "700"],
})
```

It even says `--font-inter` variable is not defined on the root element
<img width="1049" height="724" alt="image" src="https://github.com/user-attachments/assets/bbeede1d-fc85-4636-bcef-f084138ad834" />

**Reply by ackvf** [reply]
So, I also figured what was wrong, I was using
```
 className={`${inter.className} ...`}
```
but I should have used 
```
 className={`${inter.variable} ...`}
```

/edit

But I have another problem. This will not define the variables on the `:root` element, breaking tailwind.

/edit solved
https://stackoverflow.com/questions/79863796/how-to-correctly-define-next-font-in-next-js-pages-router-with-tailwind-v4s-in

**Comment by nareshbhatia** [comment, marked as answer]
I figured out the issue. Another CSS file was defining `--font-family-mono` and `--font-family-sans` as hard coded values – `Roboto Mono` and `Inter`. These were overriding the values generated by Next.js.
